### ISSUE TITLE
`ignoreRedirects` fails due to missing host in the certificate's altname

### ISSUE THREAD

**Issue by amimas** [textlint-rule/textlint-rule-no-dead-link#125]
Hi,

I was using the version `4.4.1` and trying to upgrade to the latest version (`4.6.1`). After the upgrade, I'm getting the following link being reported as dead:

https://www.consul.io/intro/getting-started/kv.html

The above link is being redirected to:

https://learn.hashicorp.com/consul/getting-started/kv.html

I already have `ignoreRedirects` to `true` and other links that are being redirected are ignored properly. The error returned by textlint is:

```log
 13:124  error  https://www.consul.io/intro/getting-started/kv.html is dead. (request to https://learn.hashicorp.com/consul/getting-started/kv.html failed, reason: Hostname/IP does not match certificate's altnames: Host: www.consul.io. is not in the cert's altnames: DNS:*.netlify.com, DNS:netlify.com)  no-dead-link
```

When I use a curl command to test that link, I get a `301` redirect status code and no other errors:

```sh
$ curl -I -H "Host: www.consul.io" -H "User-Agent: textlint-no-dead-link" -H "Accept: */*" https://www.consul.io/intro/getting-started/kv.html
HTTP/2 301 
retry-after: 0
location: https://learn.hashicorp.com/consul/getting-started/kv.html
strict-transport-security: max-age=31536000; includeSubdomains; preload
accept-ranges: bytes
date: Wed, 27 Nov 2019 18:04:42 GMT
content-length: 0
```

It seems the issue didn't exist when I was using version `4.4.1`.

**Comment by azu** [comment]
https://github.com/textlint-rule/textlint-rule-no-dead-link/releases/tag/4.6.2 will fix this issue.

**Comment by amimas** [comment]
Thanks @azu for the quick fix. I will try to test it out soon. I do have a question about the fix though.

In the PR #126, if I'm not mistaken, I noticed the host name is being updated to the new URL that is being redirected to. I'm curious why we need to that in the first place.

If I marked `https://example.com` to be ignored if it's being redirected, why does the linter need to validate new url (`https://new.example.com`)? As soon as we get a `30x` status code from the original URL, why don't we just stop there?

Did I mis-interpret the changes in that PR?
